AK/Error.h

```diff
@@ -152,33 +152,50 @@ class [[nodiscard]] ErrorOr {
     }
 
     ALWAYS_INLINE constexpr ErrorOr(ErrorOr&& other)
-    requires(!Detail::IsTriviallyMoveConstructible<T> || !Detail::IsTriviallyMoveConstructible<E>)
+    requires(!IsTriviallyMoveConstructible<T> || !IsTriviallyMoveConstructible<E>)
         : m_is_error(other.m_is_error)
     {
         if (other.is_error())
-            construct_at<E>(&m_error, other.release_error());
+            construct_at<E>(&m_error, forward<ErrorOr>(other).release_error());
```

Comment:
I don't think `other` is a [forwarding reference](https://en.cppreference.com/w/cpp/language/reference.html#Forwarding_references) here, as we're not in a template method. Does `forward` even do anything here?

Reply:
I think in this case it's a `move`?
But can make it an explicit `move` in case it is really needed